sechub-doc/src/docs/asciidoc/documents/shared/configuration/executor_config.adoc

```diff
@@ -0,0 +1,229 @@
+breaking change the exisitng product executor will be extended and the existing version will be kept.
+
+When there are breaking changes - e.g. the communication between {sechub} and a new version of the security
+product becomes completely different, or parameters have a complete other meaining, an additional
+product executor with another version will be available.
+
+===== Product base URL
+Here we define the base URL to the security product including port number.
+
+As an example: `https://checkmarx.example.org:6011`
+
+===== Product credentials 
+We need dedicated credentials to communicate/use a security product.
+
+====== User
+Just the user name for product. You can either define here the real user 
+name or use `env:ENVIRONMENT_VARIABLENAME` instead. The environment name
+must be set at SecHub server side with origin user name so the server will
+store `env:ENVIRONMENT_VARIABLENAME` as configuration in database, but will
+use the content of environment variable `ENVIRONMENT_VARIABLENAME` at 
+job execution runtime.
+
+====== Password or API token
+Same situation as for user names, described before: Use `env:ENVIRONMENT_VARIABLENAME`
+to prevent storage of real credentials in database and provide the secret information
+at server side as an environment variable.
+
+WARNING: Please use always the `env:ENVIRONMENT_VARIABLENAME` approach  - at least for passwords or api tokens 
+
+
+===== Parameters
+Parameters are used at job execution runtime to setup product adapters and to change security
+product behaviours.
+
+====== Mappings
+Some product executors do need special mappings - e.g for new project creations. Please look at
+<<mapping-technical-details, mapping>> for more details about the
+concept and the format. 
+
+====== Key values
+It is also possible to provide just dedicated key value parameters.
+
+ifdef::usecasedoc[]
+==== Usecases
+Following usecases are supported:
+
+- <<section-usecase-UC_047,UC_047-Admin creates an executor configuration>>
+
+- <<section-usecase-UC_048,UC_048-Admin deletes executor configuration>>
+
+- <<section-usecase-UC_049,UC_049-Admin fetches executor configuration list>>
+
+- <<section-usecase-UC_050,UC_050-Admin fetches executor configuration>>
+
+- <<section-usecase-UC_051,UC_051-Admin updates executor configuration>>
+endif::usecasedoc[]
+
+==== Product specific details
+
+===== Checkmarx
+====== Mapping teams and presets for new projects
+When a user starts a scan and a `checkmarx` project does not already exist, it will be created.
+So far so good.
+
+But maybe it is shall not be created with the default Checkmarx "preset" because in some scenarios this is not very suitable
+- e.g. when creating many projects, we do not want to assign all presetIds manually to the project.
+(A preset is something like a profile in Checkmarx where scope of scan can be defined at project level).
+
+Also there should be an automated assignment to dedicated teams when
+the project is new created.
```

Comment:
Maybe, this makes more sense: `the project is newly created`.

Reply:
done